store/src/java/com/zimbra/cs/imap/ImapHandler.java

```diff
@@ -4333,18 +4339,35 @@ private boolean doSTORE(String tag, String sequenceSet, List<String> flagNames,
      */
     protected boolean doCOPY(String tag, String sequenceSet, ImapPath path, boolean byUID)
             throws IOException, ImapException {
-        checkCommandThrottle(new CopyCommand(sequenceSet, path));
-        if (!checkState(tag, State.SELECTED)) {
-            return true;
-        }
+
+        CopyCommand commandP = new CopyCommand(sequenceSet, path);
         String command = (byUID ? "UID COPY" : "COPY");
         String copyuid = "";
-
         ImapFolder i4folder = getSelectedFolder();
         if (i4folder == null) {
             throw new ImapSessionClosedException();
         }
         MailboxStore mbox = i4folder.getMailbox();
+
+        try {
+            commandP.setProcessedList(processedDeletes.get(mbox.getAccountId()));
+
+        }  catch ( ServiceException e) {
+
+        }
+
+        try {
+            checkCommandThrottle(commandP);
+        } catch (ImapThrottledException e) {
+            sendOK(tag, copyuid + command + " completed");
```

Comment:
Why are we sending completion if we are throttling them, can we add a comment above this explaining why?